src/raftstore/coprocessor/split_check/keys.rs

```diff
@@ -20,37 +22,54 @@ use super::super::{Coprocessor, KeyEntry, ObserverContext, SplitCheckObserver, S
 use super::Host;
 
 pub struct Checker {
-    max_keys: u64,
-    split_keys: u64,
-    current_keys: u64,
-    split_key: Option<Vec<u8>>,
+    max_keys_count: u64,
+    split_threshold: u64,
+    current_count: u64,
+    split_keys: Vec<Vec<u8>>,
+    batch_split_limit: u64,
 }
 
 impl Checker {
-    pub fn new(max_keys: u64, split_keys: u64) -> Checker {
+    pub fn new(max_keys_count: u64, split_threshold: u64, batch_split_limit: u64) -> Checker {
         Checker {
-            max_keys,
-            split_keys,
-            current_keys: 0,
-            split_key: None,
+            max_keys_count,
+            split_threshold,
+            current_count: 0,
+            split_keys: Vec::with_capacity(1),
```

Comment:
Why not batch split limit?

Reply:
for most cases, there is only one split key produced.

Reply:
for most cases, there is only one split key produced.